configuration/src/main/java/com/synopsys/integration/configuration/config/PropertyConfiguration.java

```diff
@@ -60,6 +63,31 @@ public PropertyConfiguration(@NotNull final List<PropertySource> orderedProperty
         this.orderedPropertySources = orderedPropertySources;
     }
 
+    public List<PropertyInfo> collectPropertyInfo(List<Property> knownProperties, boolean maskPasswords) {
```

Comment:
I don't know that I love having this be on the PropertyConfiguration. I'd also prefer it was more along the lines of getMaskedValues. But also the (original) intention behind this class was to share it with other projects and this seems too specific for this class.

Perhaps making the "should mask" be a predicate and in this case we just give a detect predicate for password and api.token. Another option would be a reporter (or summarrizer) type class that takes the PropertyConfig and KnownValues and returns the masked values.

Reply:
Delegated collection process to different class.  I like the idea of taking in a predicate so I made that adjustment as well.